app.component.ts

```diff
@@ -1,110 +0,0 @@
-import { Component, OnDestroy, OnInit } from '@angular/core';
-import {
-  combineLatest,
-  filter,
-  forkJoin,
-  map,
-  Observable,
-  Subject,
-  Subscription,
-  switchMap,
-  debounceTime,
-  distinctUntilChanged,
-  merge,
-} from 'rxjs';
-import { MockDataService } from './mock-data.service';
-
-@Component({
-  selector: 'my-app',
-  templateUrl: './app.component.html',
-  styleUrls: ['./app.component.css'],
-})
-export class AppComponent implements OnInit, OnDestroy {
-  searchTermByCharacters = new Subject<string>();
-  charactersResults$: Observable<any>;
-  planetAndCharactersResults$: Observable<any>;
-  isLoading: boolean = false;
-
-  constructor(private mockDataService: MockDataService) {}
-
-  ngOnInit(): void {
-    this.initLoadingState();
-    this.initCharacterEvents();
-  }
-
-  changeCharactersInput(element): void {
-    // 1.1. Add functionality to changeCharactersInput method. Changes searchTermByCharacters Subject value on input change.
-    const inputValue: string = element.target.value;
-    // YOUR CODE STARTS HERE
-    this.searchTermByCharacters.next(inputValue);
-    // YOUR CODE ENDS HERE
-  }
-
-  initCharacterEvents(): void {
-    // 1.2. Add API call on each user input. Use mockDataService.getCharacters - to make get request.
-
-    // 2. Since we don't want to spam our service add filter by input value and do not call API until a user enters at least 3 chars.
-
-    // 3. Add debounce to prevent API calls until user stop typing.
-
-    this.charactersResults$ = this.searchTermByCharacters.pipe(
-      // YOUR CODE STARTS HERE
-      map((value: string) => value.trim()),
-      filter((value: string) => value.length === 3),
-      debounceTime(500),
-      distinctUntilChanged(),
-      switchMap((charactersResults) =>
-        this.mockDataService.getCharacters(charactersResults)
-      )
-    );
-    // YOUR CODE ENDS HERE
-  }
-
-  loadCharactersAndPlanet(): void {
-    // 4. On clicking the button 'Load Characters And Planets', it is necessary to process two requests and combine the results of both requests into one result array. As a result, a list with the names of the characters and the names of the planets is displayed on the screen.
-    // Your code should looks like this: this.planetAndCharactersResults$ = /* Your code */
-    // YOUR CODE STARTS HERE
-
-    this.planetAndCharactersResults$ = forkJoin([
-      this.mockDataService.getPlatents(),
-      this.mockDataService.getCharacters(),
-    ]).pipe(
-      map((data) => data.reduce((result, arr) => [...result, ...arr], []))
```

Comment:
as we have only two joins then no need to use reduce we can directly use by array extraction

Reply:
updated